Source/Others/DDMathEvaluator+FORM.m

```diff
@@ -38,7 +38,9 @@ + (NSDictionary *)hyp_directoryFunctions {
 
             otherValue = (variables[expression.variable]) ?: expression.variable;
 
-            if (![baseValue isEqualToString:otherValue]) {
+            BOOL baseValueIsEqual = ([baseValue isKindOfClass:[NSString class]] &&
+                                     ![baseValue isEqualToString:otherValue]);
```

Comment:
For reference : https://github.com/davedelong/DDMathParser/wiki/Built-in-Functions